src/Raven.Client/Documents/Session/DocumentSessionAttachmentsBase.cs

```diff
@@ -27,6 +27,9 @@ public AttachmentName[] GetNames(object entity)
             if (entity == null)
                 throw new ArgumentNullException(nameof(entity));
 
+            if (entity is string)
+                throw new ArgumentException($"You need to track the entity in the session for document id {entity}.", nameof(entity));
```

Comment:
Can we be more explicit here? I've read that few times and I'm still having issues to understand what would be the action that I should take to not get this exception when this method is used.

Reply:
GetNames is the only one that has no overload by 'documentId', as opposed to other attachments' methods such as get, store, etc.

Reply:
I understand, but the exception message is not 100% clear

Reply:
Done.